_tutorials/new-component/create-a-new-component-part-3.md

```diff
@@ -0,0 +1,571 @@
+	@include oBrandDefine('o-example', 'master', (
+		'variables': (
+			'border-color': oColorsByName('slate'),
+			'background-color': oColorsByName('wheat')
+			'inverse': (
+				'text-color': oColorsByName('white'),
+				'background-color': oColorsByName('slate')
+			),
+			'b2c': (
+				'background-color': oColorsByName('org-b2c-light')
+			)
+		),
+		'supports-variants': (
+			'inverse',
+			'b2c'
+		)
+	));
+}
+
+// Add internal brand configuration.
+@if oBrandGetCurrentBrand() == 'internal' {
+	@include oBrandDefine('o-example', 'internal', (
+		'variables': (
+			'border-color': oColorsByName('slate'),
+			'background-color': oColorsByName('slate-white-5'),
+			'inverse': (
+				'text-color': oColorsByName('white'),
+				'background-color': oColorsByName('slate')
+			)
+		),
+		'supports-variants': ('inverse')
+	));
+}
+
+// Add whitelabel brand configuration.
+@if oBrandGetCurrentBrand() == 'whitelabel' {
+	@include oBrandDefine('o-example', 'whitelabel', (
+		'variables': (
+			'border-color': oColorsByName('black'),
+			'background-color': oColorsByName('white'),
+			'inverse': (
+				'text-color': oColorsByName('white'),
+				'background-color': oColorsByName('black')
+			)
+		),
+		'supports-variants': ('inverse')
+	));
+}</code></pre>
+
+The `_oExampleSupports` function we briefly mentioned earlier will return `true` if a given variant name is supported by the current brand, based on the `supports-variants` configuration we just set. For example only the `master` brand has the `b2c` theme listed under `supports-variants` so `_oExampleSupports('b2c')` will only return true when the current brand is the `master` brand.
+
+### Output Theme CSS
+
+We can now complete our theme mixin:
+- Use `_oExampleSupports` with [the Sass `@error` at-rule](https://sass-lang.com/documentation/at-rules/error) to throw an error if the theme name given is not supported by the current brand.
+- Use `_oExampleGet` to get theme values.
+- Use `oButtonsContent` to update the button styles if there is a matching [o-buttons theme](https://registry.origami.ft.com/components/o-buttons@6.0.14/readme?brand=master#themes).
+
+<pre><code class="o-syntax-highlight--scss">// src/scss/_mixins.scss
+
+@mixin oExampleAddTheme($name) {
+	// Error if an unsupported theme name is given.
+	@if not _oExampleSupports($name) {
+		@error 'The name "#{$name}" is not a supported "#{oBrandGetCurrentBrand()}" brand theme';
+	}
+
+	// Output theme css.
+	.o-example--#{$name} {
+		background: _oExampleGet('background-color', $from: $theme);
+		color: _oExampleGet('text-color', $from: $theme);
+		// Theme the o-example button using o-buttons.
+		// Only output button styles to change the theme,
+		// don't repeat styles shared by all buttons.
+		$matching-button-theme: $name == 'inverse' or $name == 'b2c';
+		@if $matching-button-theme {
+			.o-example__button {
+				@include oButtonsContent($opts: (
+					'type': 'primary',
+					'theme': $matching-button-theme
```

Comment:
huh, did this code work for you @JakeChampion? just glancing through and I think this is wrong
```suggestion
					'theme': $name
```

Reply:
ah looks like in a later example its fixed, maybe that's why we didn't notice